src/core/server/base.js

```diff
@@ -71,33 +71,42 @@ export default function(routes) {
         return res.status(404).end('Not found.');
       }
 
-      const InitialComponent = (
-        <RouterContext {...renderProps} />
-      );
-
-      const componentHTML = renderToString(InitialComponent);
-      const assets = webpackIsomorphicTools.assets();
-      const styles = Object.keys(assets.styles).map((style) =>
-       `<link href=${assets.styles[style]} rel="stylesheet" type="text/css" />`
-      ).join('\n');
-
-      const HTML = stripIndent`
-      <!DOCTYPE html>
-      <html>
-        <head>
-          <meta charset="utf-8">
-          <title>Isomorphic Redux Demo</title>
-          <meta name="viewport" content="width=device-width, initial-scale=1" />
-          ${styles}
-        </head>
-        <body>
-          <div id="react-view">${componentHTML}</div>
-          <script src="${assets.javascript.main}"></script>
-        </body>
-      </html>`;
-
-      res.header('Content-Type', 'text/html');
-      return res.end(HTML);
+      const store = createStore();
+
+      return loadOnServer({...renderProps, store}).then(() => {
+        const InitialComponent = (
+          <Provider store={store} key="provider">
+            <ReduxAsyncConnect {...renderProps} />
+          </Provider>
+        );
+
+        const componentHTML = renderToString(InitialComponent);
+        const assets = webpackIsomorphicTools.assets();
+        const styles = Object.keys(assets.styles).map((style) =>
+        `<link href=${assets.styles[style]} rel="stylesheet" type="text/css" />`
+        ).join('\n');
+
+        const HTML = stripIndent`
+        <!DOCTYPE html>
+        <html>
+          <head>
+            <meta charset="utf-8">
+            <title>Isomorphic Redux Demo</title>
+            <meta name="viewport" content="width=device-width, initial-scale=1" />
+            ${styles}
+          </head>
+          <body>
+            <div id="react-view">${componentHTML}</div>
+            <script type="application/json" id="redux-store-state">
+              ${serialize(store.getState())}
```

Comment:
It also allows serializing regular expressions and functions but since I'm using `JSON.parse()` in client.js that shouldn't be supported (which I think is good).